obspy/signal/quality_control.py

```diff
@@ -138,8 +138,8 @@ def __init__(self, files, starttime=None, endtime=None,
         # Do some sanity checks. The class only works with data from a
         # single location so we have to make sure that the existing data on
         # this object and the newly added all have the same identifier.
-        ids = set(tr.id + "." + tr.stats.mseed.dataquality for tr in self.data)
-        if len(ids) != 1:
+        ids = [t.id + "." + str(t.stats.mseed.dataquality) for t in self.data]
+        if len(set(ids)) != 1:
```

Comment:
@Jollyfant please use `tr` for traces in general whenever possible. If there is a name clash with another local variable called `tr` you can use e.g. `tr_` in the list comprehension.